projects/plugins/jetpack/_inc/lib/admin-pages/class-jetpack-redux-state-helper.php

```diff
@@ -184,6 +184,7 @@ public static function get_initial_state() {
 				'hasUpdate' => (bool) get_theme_update_available( $current_theme ),
 				'support'   => array(
 					'infinite-scroll' => current_theme_supports( 'infinite-scroll' ) || in_array( $current_theme->get_stylesheet(), $inf_scr_support_themes, true ),
+					'widgets'         => current_theme_supports( 'widgets' ),
```

Comment:
While this works, I wonder if we should use `wp_is_block_theme()` instead, so we do not get situations where a plugin adds support for widgets (like when Jetpack's Instant Search is active and creates a Jetpack Search sidebar area) for a block-based theme. What do you think?

Reply:
Thanks for your review @jeherve , I've checked that suggestion.

I see that `wp_is_block_theme()` checks for the existence of two `index.html` files - if those exist, it treats the theme as a block theme. However, the `current_theme_supports()` returns true if widgets support was intentionally added by `add_theme_support` which is called in `register_sidebar`. 

If I used `wp_is_block_theme()`, we could experience use cases with incorrect behavior e.g.:
* block theme that uses `register_sidebar()` to add a sidebar. Not sure if it is reasonable to do so, but such a theme would be a block theme with a widget support
* classic theme that doesn't `register_sidebar()`. This team wouldn't be a block theme and wouldn't have support for widgets.

I think that case with "Instant Search" enabled shouldn't be a problem as the condition already includes a check for `! isInstantSearchEnabled`. Please let me know your thoughts.

Reply:
Ok, that makes sense, thanks for looking into it! Let's go with `current_theme_supports( 'widgets' )` then.